weather.go

```diff
@@ -76,19 +78,21 @@ func CallUrl(token, location string) http.Response {
 		}
 
 		if cu.Message == "city not found" {
-			log.Print("The city cannot be found")
+			fmt.Printf("The city %scannot be found, the error code is %v \n", location, resp.StatusCode)
+			os.Exit(2)
 		}
 	}
 
-	return *resp
+	r = resp.Body
+	return r
 }
 
-func Get(resp http.Response) ([]byte, error) {
+func Get(resp io.Reader) ([]byte, error) {
 
 	var w Weather
-	var c CliResponse
+	var c Conditions
 
-	read_all, err := ioutil.ReadAll(resp.Body)
+	read_all, err := ioutil.ReadAll(resp)
 	if err != nil {
 		log.Fatal(err)
```

Comment:
Good libraries don't panic (or call `log.Fatal`, which is roughly the same thing).